### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by holmanb**
## Additional Context
Remove dead code

## Merge type
- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: cloudinit/util.py [hunk 1]
```diff
@@ -2316,10 +2316,6 @@ def make_header(comment_char="#", base="created"):
     return header
 
 
-def abs_join(base, *paths):
```
Comment (TheRealFalcon) [inline cloudinit/util.py:2319]: Why are we duplicating this multiple times now?

Comment (holmanb) [reply cloudinit/util.py:2319]: Right, I didn't bother to explain that tradeoff.

My argument here is: "packaging code, and especially packaging code that is only used for building test packages, should not be shipped as part of the cloud-init runtime".

If we agree on that point we have a few options:

1. duplicate (chosen)
2. replace this one function call with the two function calls it simplifies at callsites 
3. create a new module to store shared code

Option 2 involved more typing. Option 3 seemed like overkill for a one line function. Option 1 seemed like a reasonable tradeoff, but I will happily do 2 if preferred. Option 3 sounds like a refactor to make it worthwhile. If there are other sharable/shared functions to consolidate I guess we can do that too.

Comment (holmanb) [reply cloudinit/util.py]: @TheRealFalcon alright, per out of band convo it sounds like 2 is preferred.

**Review by TheRealFalcon — COMMENTED** [review]
Did `abs_join` get unremoved? I'm seeing it removed here.

**Comment by holmanb** [comment]
> Did `abs_join` get unremoved? I'm seeing it removed here.

Thanks, fixed!

**Review by TheRealFalcon — APPROVED** [review]

**Comment by holmanb** [comment]
force pushed the to update the commit wording with PR number

**Pull request merged**